### PR TITLE
wio: init at unstable-2023-05-28

### LINKED ISSUE #257000: Package request: wio (fork)
**Project description**

Wayland compositor similar to Plan 9's rio (mirror).
Note that the upstream for this software was [removed previously](https://github.com/NixOS/nixpkgs/pull/228632) due to being unmaintained and depending on an outdated wlroots. I am proposing that this fork of that software is added to pkgs and the entry for the original in aliases.nix is amended to point to the fork. Note also that the only changes made from the original wio derivation to build the software (other then changing `src`) is the addition of `preConfigure = ''sed -i "16 a '-Wno-maybe-uninitialized'," meson.build'';`. I am pinging @AndersonTorres because they where the maintainer of the deprecated software that this addition will posthumously replace.

**Metadata**

* homepage URL: https://github.com/Rubo3/wio
* source URL: https://github.com/Rubo3/wio
* license: BSD 3-Clause
* platforms: linux

### PR DESCRIPTION
**Pull request by AndersonTorres**
A resurrection.

Closes https://github.com/NixOS/nixpkgs/issues/257000

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/wi/wio/package.nix [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    xwayland
+  ];
+
+  strictDeps = false; # why is it so hard?
+
+  env.NIX_CFLAGS_COMPILE = "-Wno-error=maybe-uninitialized";
+
+  postInstall = ''
+    wrapProgram $out/bin/wio \
+      --prefix PATH ":" "${lib.makeBinPath [ alacritty cage ]}"
```
Comment (wineee) [inline pkgs/by-name/wi/wio/package.nix:56]: Dependence on alacritty should not be forced, it should be up to the user to decide which terminal emulator to use

File: pkgs/by-name/wi/wio/package.nix [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+, makeWrapper
+, mesa
+, meson
+, ninja
+, pkg-config
+, udev
+, unstableGitUpdater
+, wayland
+, wayland-protocols
+, wlroots
```
Comment (wineee) [inline pkgs/by-name/wi/wio/package.nix:17]: should use `wlroots_0_16`

https://github.com/NixOS/nixpkgs/pull/269359  will break it

File: pkgs/by-name/wi/wio/package.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/by-name/wi/wio/package.nix:56]: I merely resurrected the expression.

File: pkgs/by-name/wi/wio/package.nix (hunk 2 above)
Comment (AndersonTorres) [reply pkgs/by-name/wi/wio/package.nix:17]: Good to know. I will open a PR for this later.

**Pull request merged**